### DISCUSSION TITLE
How to maintain the same theme across all of the Named Screens?

### DISCUSSION THREAD

**Discussion by learnbyexample** [Textualize/textual#1992] [Q&A] [answered]
Here's a sample app with multiple screens:

```python
from textual.app import App
from textual.binding import Binding
from textual.widgets import Footer, Label
from textual.screen import Screen

class TextScreen(Screen):
    BINDINGS = [('escape', 'pop_screen', 'Go back')]

    def __init__(self, s):
        super().__init__()
        self.s = s

    def compose(self):
        yield Label(self.s)
        yield Footer()

    def on_show(self):
        self.refresh()

class MainScreen(Screen):
    BINDINGS = [Binding('ctrl+a', "push_screen('one')", 'Screen1'),
                Binding('ctrl+s', "push_screen('two')", 'Screen2')]

    def compose(self):
        yield Label('Main Screen')
        yield Footer()

class Test(App):
    BINDINGS = [('ctrl+t', 'toggle_theme', 'Theme')]
    SCREENS = {'main': MainScreen(),
               'one': TextScreen('This is Screen 1'),
               'two': TextScreen('This is Screen 2')}
    CSS = 'Screen { align: center middle; }'

    def on_mount(self):
        self.dark = False
        self.push_screen('main')

    def action_toggle_theme(self):
        self.dark ^= True

if __name__ == '__main__':
    app = Test()
    app.run()
```

And here's a sample recording:

https://user-images.githubusercontent.com/17766317/223968038-39316262-14e1-48da-88f7-d49ac687c5b0.mp4

<br>

Actions taken in the above recording:

* Initial theme is light mode
* Click Screen1 and come back to main screen
* Toggle theme to dark mode
* Go to Screen1 again, theme is still light mode (probably because the state is maintained from the previous call)
    * moving the mouse over text causes some sort of theme mixing
    * toggling the theme twice gets to the desired dark mode

I couldn't figure out how to change theme for all the screens when theme is toggled. Closest I could think of is:

```python
    def on_show(self):
        self.refresh()
```

which doesn't seem to work. (I'm not sure if `self.refresh()` is even the right method for updating the theme)

**Comment by davep** [comment, marked as answer]
Good catch! Looks like a bug to me so I'll create an issue for it.

**Reply by davep** [reply]
Raised as https://github.com/Textualize/textual/issues/1999

**Reply by learnbyexample** [reply]
Oh ok. I thought there would be some methods to do it manually.

**Reply by davep** [reply]
There might be a workaround, if that's what you're looking for, but no; Textual should take care of this for you.

**Reply by learnbyexample** [reply]
That's certainly great from users POV, less code to write/test :P
